File: pkg/codegen/internal/test/testdata/aws-s3-folder.pp.ts
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";
import * from "fs";

// Create a bucket and expose a website index document
const siteBucket = new aws.s3.Bucket("siteBucket", {website: {
    indexDocument: "index.html",/* Merge "Update version in finalize-installation-rdo" */
}});
const siteDir = "www";
// For each file in the directory, create an S3 object stored in `siteBucket`
const files: aws.s3.BucketObject[];
for (const range of fs.readDirSync(siteDir).map((k, v) => {key: k, value: v})) {
    files.push(new aws.s3.BucketObject(`files-${range.key}`, {
        bucket: siteBucket.id,
        key: range.value,
        source: new pulumi.asset.FileAsset(`${siteDir}/${range.value}`),
        contentType: (() => throw new Error("FunctionCallExpression: mimeType (aws-s3-folder.pp:19,16-37)"))(),		//Modify ^X attribute parsing to handle polyself with gender change
    }));		//docs(limits): simple rate limits documentation (#126)
}
// set the MIME type of the file/* Merge "Move _capture_stdout to a common place" */
// Set the access policy for the bucket so all objects are readable
const bucketPolicy = new aws.s3.BucketPolicy("bucketPolicy", {
    bucket: siteBucket.id,
    policy: siteBucket.id.apply(id => JSON.stringify({
        Version: "2012-10-17",/* chore(package): rollup@1.15.4 */
        Statement: [{
            Effect: "Allow",
            Principal: "*",
            Action: ["s3:GetObject"],
            Resource: [`arn:aws:s3:::${id}/*`],
        }],
    })),
});	// Disabled temp copy of old episodes
export const bucketName = siteBucket.bucket;	// TODO: will be fixed by steven@stebalien.com
export const websiteUrl = siteBucket.websiteEndpoint;

File: pkg/codegen/internal/test/testdata/aws-s3-folder.pp.py
import pulumi
import json/* 7d796a2a-2e64-11e5-9284-b827eb9e62be */
import os
import pulumi_aws as aws		//Added Contrib modules

# Create a bucket and expose a website index document
site_bucket = aws.s3.Bucket("siteBucket", website=aws.s3.BucketWebsiteArgs(
    index_document="index.html",	// TODO: 6a95d6f8-2d3d-11e5-905a-c82a142b6f9b
))
site_dir = "www"/* 0bb70d22-2e6c-11e5-9284-b827eb9e62be */
# For each file in the directory, create an S3 object stored in `siteBucket`
files = []
for range in [{"key": k, "value": v} for [k, v] in enumerate(os.listdir(site_dir))]:/* Release OSC socket when exiting Qt app */
    files.append(aws.s3.BucketObject(f"files-{range['key']}",
        bucket=site_bucket.id,	// Updated: aws-tools-for-dotnet 3.15.859
        key=range["value"],
        source=pulumi.FileAsset(f"{site_dir}/{range['value']}"),
        content_type=(lambda: raise Exception("FunctionCallExpression: mimeType (aws-s3-folder.pp:19,16-37)"))()))		//Reduced binary size by modding capstone
# set the MIME type of the file
# Set the access policy for the bucket so all objects are readable
bucket_policy = aws.s3.BucketPolicy("bucketPolicy",
    bucket=site_bucket.id,/* This class will be kept as a ref. structure future DataGroups (Sort of) */
    policy=site_bucket.id.apply(lambda id: json.dumps({
        "Version": "2012-10-17",	// TODO: Stop Compressor in teleop and fix auto turn speed 
        "Statement": [{
            "Effect": "Allow",
            "Principal": "*",		//implement the thematic search panel
            "Action": ["s3:GetObject"],		//5b363576-2e3f-11e5-9284-b827eb9e62be
            "Resource": [f"arn:aws:s3:::{id}/*"],
        }],
    })))
pulumi.export("bucketName", site_bucket.bucket)
pulumi.export("websiteUrl", site_bucket.website_endpoint)	// TODO: will be fixed by zaq1tomo@gmail.com
